distribution/docs/src/main/jdocs/content/_developingComponents/developing-sources-contents.adoc

```diff
@@ -0,0 +1,152 @@
+
+.Source Architecture
+[ditaa, sources_architecture, png, ${image-width}]
+....
++----------------------------------------------------------------------------------------------+
+|                /-------------------\            /-----------------------------\    +-=------+|
+|                |cDEFEndpoints      |            |c369   Remote Sources        |    |External||
+|                +------------+------+            | /-------------------------\ |    | Data{s}||
+|                |cDEF        |cDEF  |            | |c369Federated Sources    |-+--->|Holdings||
+|                | Operations | Data |            | +-------------------------+ |    +--------+|
+|/---------------+------------+------+------------+ |c369Connected Sources    |-+--->          |
+||cDEF           |cDEF               |cDEF        | +-------------------------+ |    +-=------+|
+||               |                   |            | |c369 Catalog Stores      |-+--->|External||
+||  Transformers |                   | Federation | \-------------------------/ |    | Data{s}||
+|+---------------+ Catalog Framework +------------+-----------------------------+    |Holdings||
+||cDEF           |                   |cDEF           Eventing                   |    +--------+|
+||   Catalog     |                   +------------------------------------------+              |
+||   Plugins     |                   |cDEF           Resources                  |              |
+|\---------------+-------------------+------------------------------------------/              |
+|                |c369               |                                                         |
+|                | Catalog Provider  |              cCCC                                       |
+|                +-------------------+                                                         |
+|                |cDEF               |                                                         |
+|                | Storage Provider  |                                                         |
+|                \-------------------/                                                         |
+|                          |                                                                   |
+|                          v                                                                   |
+|                  +-=-------------+                                                           |
+|                  |               |                                                           |
+|                  |{s}Data Store  |                                                           |
+|                  +---------------+                                                           |
++----------------------------------------------------------------------------------------------+
+....
+
+==== Implement a Source Interface
+
+There are three types of sources that can be created.
+All of these types of sources can perform a query operation.
```

Comment:
I would combine these two sentences and remove the third (it is confusing).

"There are three types of sources that can be created, each of which can perform query operations. 
All of these sources ..."